arm-bsp/firmware-image-juno: add xen support

This patch adds xen and dtbs binaries entries to images-r[012].txt files.
These images-r[012].txt files contain NOR filesystem entries details,
like file offset, name, path etc.

It also adds customization for uEnv.txt file, that allows to autoboot xen.

Upstream-Status: Inappropriate [configuration]
Signed-off-by: Kamil Dziezyk <kamil.dziezyk@arm.com>

diff -u a/images-r0.txt b/images-r0.txt
--- a/images-r0.txt	2020-11-25 20:25:38.677687712 +0100
+++ b/images-r0.txt	2020-11-25 20:36:55.482194294 +0100
@@ -1,7 +1,7 @@
 TITLE: Versatile Express Images Configuration File
 
 [IMAGES]
-TOTALIMAGES: 10                  ;Number of Images (Max: 32)
+TOTALIMAGES: 11                  ;Number of Images (Max: 32)
 
 NOR0UPDATE: AUTO                 ;Image Update:NONE/AUTO/FORCE
 NOR0ADDRESS: 0x00000000          ;Image Flash Address
@@ -17,14 +17,14 @@
 
 NOR2UPDATE: AUTO                 ;Image Update:NONE/AUTO/FORCE
 NOR2ADDRESS: 0x00500000          ;Image Flash Address
-NOR2FILE: \SOFTWARE\Image        ;Image File Name
-NOR2NAME: norkern                ;Rename kernel to norkern
+NOR2FILE: \SOFTWARE\Image.lzm    ;Image File Name
+NOR2NAME: norkern.lzm            ;Rename kernel to norkern.lzm
 NOR2LOAD: 00000000               ;Image Load Address
 NOR2ENTRY: 00000000              ;Image Entry Point
 
 NOR3UPDATE: AUTO                 ;Image Update:NONE/AUTO/FORCE
 NOR3ADDRESS: 0x02900000          ;Image Flash Address
-NOR3FILE: \SOFTWARE\juno.dtb     ;Image File Name
+NOR3FILE: \SOFTWARE\XEN\juno.dtb ;Image File Name
 NOR3NAME: board.dtb              ;Specify target filename to preserve file extension
 NOR3LOAD: 00000000               ;Image Load Address
 NOR3ENTRY: 00000000              ;Image Entry Point
@@ -64,8 +64,13 @@
 NOR9UPDATE: AUTO                 ;Image Update:NONE/AUTO/FORCE
 NOR9ADDRESS: 0x02A80000          ;Image Flash Address
 NOR9NAME: uEnv.txt
-NOR9FILE: \SOFTWARE\uEnv.txt     ;Image File Name
+NOR9FILE: \SOFTWARE\uenvfile     ;Image File Name
 NOR9LOAD: 00000000               ;Image Load Address
 NOR9ENTRY: 00000000              ;Image Entry Point
 
-
+NOR10UPDATE: AUTO                ;Image Update:NONE/AUTO/FORCE
+NOR10ADDRESS: 0x03000000         ;Image Flash Address
+NOR10FILE: \SOFTWARE\XEN\xen     ;Image File Name
+NOR10NAME: xen
+NOR10LOAD: 00000000              ;Image Load Address
+NOR10ENTRY: 00000000             ;Image Entry Point
diff -u a/images-r1.txt b/images-r1.txt
--- a/images-r1.txt	2020-11-25 20:40:19.005177152 +0100
+++ b/images-r1.txt	2020-11-25 20:41:17.500886263 +0100
@@ -1,7 +1,7 @@
 TITLE: Versatile Express Images Configuration File
 
 [IMAGES]
-TOTALIMAGES: 10                  ;Number of Images (Max: 32)
+TOTALIMAGES: 11                  ;Number of Images (Max: 32)
 
 NOR0UPDATE: AUTO                 ;Image Update:NONE/AUTO/FORCE
 NOR0ADDRESS: 0x00000000          ;Image Flash Address
@@ -17,14 +17,14 @@
 
 NOR2UPDATE: AUTO                 ;Image Update:NONE/AUTO/FORCE
 NOR2ADDRESS: 0x00500000          ;Image Flash Address
-NOR2FILE: \SOFTWARE\Image        ;Image File Name
-NOR2NAME: norkern                ;Rename kernel to norkern
+NOR2FILE: \SOFTWARE\Image.lzm    ;Image File Name
+NOR2NAME: norkern.lzm            ;Rename kernel to norkern.lzm
 NOR2LOAD: 00000000               ;Image Load Address
 NOR2ENTRY: 00000000              ;Image Entry Point
 
 NOR3UPDATE: AUTO                 ;Image Update:NONE/AUTO/FORCE
 NOR3ADDRESS: 0x02900000          ;Image Flash Address
-NOR3FILE: \SOFTWARE\juno-r1.dtb  ;Image File Name
+NOR3FILE: \SOFTWARE\XEN\juno-r1.dtb ;Image File Name
 NOR3NAME: board.dtb              ;Specify target filename to preserve file extension
 NOR3LOAD: 00000000               ;Image Load Address
 NOR3ENTRY: 00000000              ;Image Entry Point
@@ -64,8 +64,13 @@
 NOR9UPDATE: AUTO                 ;Image Update:NONE/AUTO/FORCE
 NOR9ADDRESS: 0x02A80000          ;Image Flash Address
 NOR9NAME: uEnv.txt
-NOR9FILE: \SOFTWARE\uEnv.txt     ;Image File Name
+NOR9FILE: \SOFTWARE\uenvfile     ;Image File Name
 NOR9LOAD: 00000000               ;Image Load Address
 NOR9ENTRY: 00000000              ;Image Entry Point
 
-
+NOR10UPDATE: AUTO                ;Image Update:NONE/AUTO/FORCE
+NOR10ADDRESS: 0x03000000         ;Image Flash Address
+NOR10FILE: \SOFTWARE\XEN\xen     ;Image File Name
+NOR10NAME: xen
+NOR10LOAD: 00000000              ;Image Load Address
+NOR10ENTRY: 00000000             ;Image Entry Point
diff -u a/images-r2.txt b/images-r2.txt
--- a/images-r2.txt	2020-11-25 20:40:30.625119321 +0100
+++ b/images-r2.txt	2020-11-25 20:41:30.720820597 +0100
@@ -1,7 +1,7 @@
 TITLE: Versatile Express Images Configuration File
 
 [IMAGES]
-TOTALIMAGES: 10                  ;Number of Images (Max: 32)
+TOTALIMAGES: 11                  ;Number of Images (Max: 32)
 
 NOR0UPDATE: AUTO                 ;Image Update:NONE/AUTO/FORCE
 NOR0ADDRESS: 0x00000000          ;Image Flash Address
@@ -17,14 +17,14 @@
 
 NOR2UPDATE: AUTO                 ;Image Update:NONE/AUTO/FORCE
 NOR2ADDRESS: 0x00500000          ;Image Flash Address
-NOR2FILE: \SOFTWARE\Image        ;Image File Name
-NOR2NAME: norkern                ;Rename kernel to norkern
+NOR2FILE: \SOFTWARE\Image.lzm    ;Image File Name
+NOR2NAME: norkern.lzm            ;Rename kernel to norkern.lzm
 NOR2LOAD: 00000000               ;Image Load Address
 NOR2ENTRY: 00000000              ;Image Entry Point
 
 NOR3UPDATE: AUTO                 ;Image Update:NONE/AUTO/FORCE
 NOR3ADDRESS: 0x02900000          ;Image Flash Address
-NOR3FILE: \SOFTWARE\juno-r2.dtb  ;Image File Name
+NOR3FILE: \SOFTWARE\XEN\juno-r2.dtb ;Image File Name
 NOR3NAME: board.dtb              ;Specify target filename to preserve file extension
 NOR3LOAD: 00000000               ;Image Load Address
 NOR3ENTRY: 00000000              ;Image Entry Point
@@ -64,8 +64,13 @@
 NOR9UPDATE: AUTO                 ;Image Update:NONE/AUTO/FORCE
 NOR9ADDRESS: 0x02A80000          ;Image Flash Address
 NOR9NAME: uEnv.txt
-NOR9FILE: \SOFTWARE\uEnv.txt     ;Image File Name
+NOR9FILE: \SOFTWARE\uenvfile     ;Image File Name
 NOR9LOAD: 00000000               ;Image Load Address
 NOR9ENTRY: 00000000              ;Image Entry Point
 
-
+NOR10UPDATE: AUTO                ;Image Update:NONE/AUTO/FORCE
+NOR10ADDRESS: 0x03000000         ;Image Flash Address
+NOR10FILE: \SOFTWARE\XEN\xen     ;Image File Name
+NOR10NAME: xen
+NOR10LOAD: 00000000              ;Image Load Address
+NOR10ENTRY: 00000000             ;Image Entry Point
diff -u a/uEnv.txt b/uEnv.txt
--- a/uEnv.txt	2020-11-20 13:48:31.845078690 +0100
+++ b/uEnv.txt	2020-11-25 20:30:00.572306675 +0100
@@ -1,11 +1,16 @@
+xen_name=xen
+xen_addr=0x84000000
 uenvcmd=run mybootcmd
+kernel_alt_name=norkern.lzm
+kernel_comp_addr_r=0x88080000
 mybootcmd=echo Loading custom boot command; \
 echo Loading kernel; \
 afs load ${kernel_name} ${kernel_addr_r} ; \
-if test $? -eq 1; then echo Loading ${kernel_alt_name} instead of ${kernel_name}; afs load ${kernel_alt_name} ${kernel_addr_r}; fi; \
+if test $? -eq 1; then echo Loading ${kernel_alt_name} instead of ${kernel_name}; afs load ${kernel_alt_name} ${kernel_comp_addr_r}; lzmadec ${kernel_comp_addr_r} ${kernel_addr_r}; fi; \
 echo Loading device tree; \
 afs load  ${fdtfile} ${fdt_addr_r}; \
 if test $? -eq 1; then echo Loading ${fdt_alt_name} instead of ${fdtfile}; \
 afs load ${fdt_alt_name} ${fdt_addr_r}; fi; fdt addr ${fdt_addr_r}; fdt resize; \
-booti ${kernel_addr_r} - ${fdt_addr_r};
-
+echo Loading Xen; \
+afs load  ${xen_name} ${xen_addr}; \
+if test $? -eq 0; then echo Booting Xen; bootefi ${xen_addr} ${fdt_addr_r}; fi;

